### PR TITLE
#27 added tests for the Protocol class

### PR DESCRIPTION
**Pull request by cosmicnet**
Automated testing for the Protocol class along with a few updates to the Protocol class itself

### CODE REVIEW

**Review by jewh — CHANGES_REQUESTED** [review]
Looks fantastic, just wondering if lines 30-34 need to be repeated in lines 36-40? 

Happy to merge if that's intentional.

File: pkmodel/tests/test_protocol.py [hunk 1]
```diff
@@ -10,5 +10,71 @@ def test_create(self):
...
+            protocol.Initial_dose
+        with self.assertRaises(AttributeError):
+            protocol.Time_span
+        with self.assertRaises(AttributeError):
+            protocol._Name
+        with self.assertRaises(AttributeError):
+            protocol._Initial_dose
+        with self.assertRaises(AttributeError):
+            protocol._Time_span
+        with self.assertRaises(AttributeError):
```
Comment (jewh) [inline pkmodel/tests/test_protocol.py:35]: Don't lines 35-40 just repeat lines 29-34?

Comment (cosmicnet) [reply pkmodel/tests/test_protocol.py:35]: One sequence is trying to access private variables starting with a single _.
The other is trying to access private variables starting with a double __.

**Comment by cosmicnet** [comment]
A class variable that starts with a single _ in python is a notation for saying "This variable should be private to this class and not accessed directly, bypassing and methods to modify it such as getters and setters". However, it is still possible for a user to adjust the variable directly.
Later versions of python introduced the double __ notation for saying the same thing, but actually preventing (or making it very hard at least) the variable from being edited directly so that users can only modify it through defined methods.

File: pkmodel/tests/test_protocol.py (hunk 1 above)
Comment (jewh) [reply pkmodel/tests/test_protocol.py:35]: Ah makes sense, no issues then. I'll approve the merge.

**Pull request merged**